src/cli.ts

```
@@ -960,22 +968,43 @@ export const main = async (argv: any, version?: string) => {
+                        "latest-success-rate": record.progress.successRate / 100,
+                        "selection-weight": record.progress.selectionWeight,
                     });
                     record.reset();
                     span.end();
                 });
             }
+            // report write rpcs performance
+            if (state.writeRpc) {
+                for (const rpc in state.writeRpc.metrics) {
+                    await tracer.startActiveSpan("rpc-report", {}, roundCtx, async (span) => {
+                        const record = state.writeRpc!.metrics[rpc];
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)